src/main/java/wolox/training/models/User.java

```diff
@@ -58,23 +57,23 @@ public String getUsername() {
     }
 
     private void setUsername(String username) {
-        this.username = username;
+        this.username = checkNotNull(username);
     }
 
     public String getName() {
         return name;
     }
 
     private void setName(String name) {
-        this.name = name;
+        this.name = checkNotNull(name);
     }
 
     public LocalDate getBirthday() {
         return birthday;
     }
 
     private void setBirthday(LocalDate birthday) {
-        this.birthday = birthday;
+        this.birthday = checkNotNull(birthday);
```

Comment:
After checking if it is null you can check if the date is later than the current one.